rest/admin_api.go

```diff
@@ -818,3 +818,92 @@ func (h *handler) handlePurge() error {
 
 	return nil
 }
+
+// sg-replicate endpoints
+func (h *handler) getReplications() error {
+	replications, err := h.db.SGReplicateMgr.GetReplications()
+	if err != nil {
+		return err
+	}
+	bytes, err := base.JSONMarshal(replications)
+	h.writeRawJSON(bytes)
+	return err
+}
+
+func (h *handler) getReplication() error {
+	replicationID := mux.Vars(h.rq)["replicationID"]
+	replication, err := h.db.SGReplicateMgr.GetReplication(replicationID)
+	if replication == nil {
+		if err == nil {
+			return kNotFoundError
+		}
+		return err
+	}
+
+	bytes, err := base.JSONMarshal(replication)
+	h.writeRawJSON(bytes)
+	return err
+}
+
+func (h *handler) putReplication() error {
+	replicationID := mux.Vars(h.rq)["replicationID"]
+	return h.upsertReplication(replicationID)
+}
+
+func (h *handler) upsertReplication(replicationID string) error {
```

Comment:
Just for my own benefit, is this split out from putReplication to be used for the future backwards compatibility on the `_replicate` endpoint?

Reply:
Ah, thanks for catching.  That is legacy from when PUT and POST were invoking different handler methods.  Now that they are all going through putReplication, I should avoid the extra call.